### PR TITLE
Convert platform templates to use hooks.

### PR DESCRIPTION
**Pull request by lgalis**
Convert the platform template page to use hooks.  Update filtering and specs.

https://projects.engineering.redhat.com/browse/SSP-907

### CODE REVIEW

File: src/smart-components/platform/platform-templates.js [hunk 1]
```diff
@@ -1,133 +1,163 @@
...
-      items: this.props.platformItems
-      .filter(item => filterServiceOffering(item, this.state.filterValue))
-      .map(data => <PlatformItem key={ data.id } { ...data } />),
-      isLoading: this.props.isPlatformDataLoading
+      items: data ? data.map(item => <PlatformItem key={ item.id } { ...item } />) : [],
+      isLoading: isFetching || isFiltering
     };
 
-    let title = this.props.platform ? this.props.platform.name : '';
+    let title = platform ? platform.name : '';
```
Comment (Hyperkid123) [inline src/smart-components/platform/platform-templates.js]: this should be `const`

We should add linter rule for this.

File: src/smart-components/platform/platform-templates.js [hunk 2]
```diff
@@ -1,133 +1,163 @@
...
+            numberOfItems: paginationCurrent.count,
+            onPerPageSelect: handleOnPerPageSelect,
+            page: getCurrentPage(paginationCurrent.limit, paginationCurrent.offset),
+            onSetPage: handleSetPage,
             direction: 'down'
           }
         }) }/>
-        <ContentGallery { ...filteredItems }/>
+        <ContentGallery title={ title }
+          isLoading={ isFetching || isFiltering }
```
Comment (Hyperkid123) [inline src/smart-components/platform/platform-templates.js]: This is a duplicate props. It is also present in the filtered items.

I think it should be removed from `filteredItems` and assign both props explicitly, not by using the spread operator.

File: src/smart-components/platform/platform-templates.js [hunk 3]
```diff
@@ -1,133 +1,163 @@
...
-    platform: selectedPlatform,
-    platformItems: platformItemsData && platformItemsData.data,
-    isPlatformDataLoading
-  };
+    );};
+
+  return (
+    <Switch>
+      <Route path={ '/platforms/detail/:id/platform-templates' }
+        render={ renderItems } />
```
Comment (Hyperkid123) [inline src/smart-components/platform/platform-templates.js]: unnecessary new line

File: src/smart-components/platform/platform-templates.js [hunk 4]
```diff
@@ -1,133 +1,163 @@
...
-    paginationLinks: platformItemsData && platformItemsData.links,
-    paginationCurrent: platformItemsData && platformItemsData.meta,
-    platform: selectedPlatform,
-    platformItems: platformItemsData && platformItemsData.data,
-    isPlatformDataLoading
-  };
+    );};
+
+  return (
+    <Switch>
```
Comment (Hyperkid123) [inline src/smart-components/platform/platform-templates.js]: If there is only one child route, there is no need for `Switch`

File: src/smart-components/platform/platform-templates.js [hunk 5]
```diff
@@ -1,133 +1,163 @@
...
 };
 
-const mapDispatchToProps = dispatch => bindActionCreators({
-  fetchSelectedPlatform,
-  fetchPlatformItems
-}, dispatch);
-
 PlatformTemplates.propTypes = {
   filteredItems: PropTypes.object,
   isPlatformDataLoading: PropTypes.bool,
```
Comment (Hyperkid123) [inline src/smart-components/platform/platform-templates.js]: This props is no loner used

File: src/smart-components/platform/platform-templates.js [hunk 6]
```diff
@@ -1,133 +1,163 @@
...
-    count: PropTypes.number.isRequired
+    count: PropTypes.number
   })
 };
 
 PlatformTemplates.defaultProps = {
   platformItems: [],
   paginationCurrent: {
-    limit: 50
+    limit: 50,
```
Comment (Hyperkid123) [inline src/smart-components/platform/platform-templates.js]: Prop types which came from `mapStateToProps`  are no longer useful.

File: src/smart-components/platform/platform.js [hunk 7]
```diff
@@ -1,48 +1,31 @@
...
   return (
     <Fragment>
       <Switch>
-        <Route path={ `/platforms/detail/:id/platform-templates` } component={ PlatformTemplates }/>
-        <Route path={ `/platforms/detail/:id/platform-inventories` } component={ PlatformInventories }/>
-        <Route render={ () => <Redirect to={ `/platforms/detail/${props.match.params.id}/platform-templates` } /> } />
+        <Route path={ `/platforms/detail/:id/platform-templates` }>
+          <PlatformTemplates/>
+        </Route>
+        <Route path={ `/platforms/detail/:id/platform-inventories` }>
```
Comment (Hyperkid123) [inline src/smart-components/platform/platform.js]: Can we use the `component`/`render` syntax? So it matches the rest of the application.

Comment (lgalis) [reply src/smart-components/platform/platform.js]: @Hyperkid123  - I think we should start using this syntax, to 'stay ahead of the curve' - https://reacttraining.com/blog/react-router-v5-1/...

**Pull request merged**